src/modules/app/browser-init.service.ts

```diff
@@ -134,4 +137,23 @@ export class BrowserInitService extends InitService {
   protected initGoogleAnalytics() {
     this.googleAnalyticsService.addTrackingIdToPage();
   }
+
+  /**
+   * During the external authentication flow invalidates the SSR transferState
+   * data in the cache. This allows the app to fetch fresh content.
+   * @private
+   */
+  private externalAuthCheck() {
+
+    this.authService.isExternalAuthentication().pipe(
+        filter((externalAuth: boolean) => externalAuth)
+      ).subscribe(() => {
```

Comment:
The way this is implemented, the subscription stays open indefinitely. Is there a way to implement it so we auto-unsubscribe once we know whether or not it was an external authentication?

Reply:
Good point. Perhaps I can set a reference to the subscription and unsubscribe() when authenticationReady$() emits. I'll give it a look.

Reply:
I added an `unsubscribe()` after `firstValueFrom(this.authenticationReady$())` returns. Since this always emits a value the subscription should always close.

Reply:
`firstValueFrom` seems to be the updated version of `toPromise()`